keel-core/src/main/kotlin/com/netflix/spinnaker/keel/lifecycle/LifecycleEvent.kt

```diff
@@ -0,0 +1,28 @@
+package com.netflix.spinnaker.keel.lifecycle
+
+import java.time.Instant
+
+data class LifecycleEvent(
+  val scope: LifecycleEventScope,
+  val artifactRef: String,
+  val artifactVersion: String,
+  val type: LifecycleEventType,
+  val id: String? = null,
```

Comment:
Does the code contain any events yet with id of null? I didn't see any. If not, I'd suggest making this field required for now, and not nullable in the database, and when we hit a use case with null id events, we can add it then.

Reply:
Seems reasonable. I'll do it.